lib/src/features/symbol_line_renderer.dart

```diff
@@ -17,55 +14,80 @@ class SymbolLineRenderer extends FeatureRenderer {
   SymbolLineRenderer(this.logger);
 
   @override
-  void render(Context context, ThemeLayerType layerType, Style style,
-      TileLayer layer, TileFeature feature) {
+  void render(
+    FeatureRendererContext context,
+    ThemeLayerType layerType,
+    Style style,
+    TileLayer layer,
+    TileFeature feature,
+  ) {
     final textPaint = style.textPaint;
     final textLayout = style.textLayout;
     if (textPaint == null || textLayout == null) {
       logger.warn(() => 'line symbol does not have a text paint or layout');
       return;
     }
 
-    final lines = feature.lines;
-    logger.log(() => 'rendering linestring symbol');
+    // What if the feature has multiple paths?
+    final path = feature.paths.first;
+    if (!context.isPathWithinTileClip(path)) {
+      return;
+    }
+
+    final metrics = path.computeMetrics().toList();
+    if (metrics.length == 0) {
+      return;
+    }
+
     final evaluationContext = EvaluationContext(
-        () => feature.properties, feature.type, context.zoom, logger);
+      () => feature.properties,
+      feature.type,
+      context.zoom,
+      logger,
+    );
+
     final text = textLayout.text.evaluate(evaluationContext);
-    if (text != null) {
-      final path = Path();
-      for (final line in lines) {
-        path.addPolygon(line.toPoints(layer.extent, tileSize), false);
-      }
-      if (!_isWithinClip(context, path)) {
-        return;
+    if (text == null) {
+      return;
+    }
+
+    final textAbbr = TextAbbreviator().abbreviate(text);
```

Comment:
I'd prefer to spell things out in full to avoid creating a barrier to others reading the code. We should go wth `text` or `textAbbreviation`